net/art/table.go

```diff
@@ -11,3 +11,152 @@
 // For more information, see Yoichi Hariguchi's paper:
 // https://cseweb.ucsd.edu//~varghese/TEACH/cs228/artlookup.pdf
 package art
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/netip"
+	"strings"
+)
+
+// Table is an IPv4 and IPv6 routing table.
+type Table[T any] struct {
+	v4 strideTable[T]
+	v6 strideTable[T]
+}
+
+// Get does a route lookup for addr and returns the associated value, or nil if
+// no route matched.
+func (t *Table[T]) Get(addr netip.Addr) *T {
```

Comment:
not `(_ T, ok bool)` ?

Reply:
Why? We're storing a pointer to a route entry, it seems unnecessary to deref it before returning. I'm anticipating T will be a Peer-sized struct.

Reply:
In hindsight, with the last couple iterations of stuff, I could store a T rather than a *T, and leave the pointerification decision to the consumer - although the costs are very non-linear as sizeof(T) changes (512 * sizeof(T) * however many stride tables are alive in the Table), so that feels like a bit of a footgun to me?